.chronus/changes/tsp-openapi3-add-context-2024-7-20-9-44-3.md

```diff
@@ -0,0 +1,7 @@
+---
+changeKind: fix
+packages:
+  - "@typespec/openapi3"
+---
+
+Fixes issue in tsp-openapi3 that resulted in component schemas and parameters with the same name being merged into a single TypeSpec data type.
```

Comment:
I think this is the other pr, isn't it?  I assume this is dependent on that PR

Reply:
Yes - this branch builds off the last commit on the other PR. Once that's merged I'll rebase this as necessary and mark it ready for review.